### PR TITLE
MINIFICPP-1555 Improve the coverage of GetFileTests

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1555

-------------------------------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/standard-processors/tests/unit/GetFileTests.cpp [hunk 1]
```diff
@@ -140,37 +142,103 @@ TEST_CASE("GetFile keeps the source file if KeepSourceFile is true") {
...
-  plan->runNextProcessor();
-  plan->runNextProcessor();
+TEST_CASE("Test file filtering of GetFile", "[getFileProperty]") {
+  GetFileTestController test_controller;
+  test_controller.setProperty(minifi::processors::GetFile::FileFilter, ".?test\\.txt$");
+  test_controller.setProperty(minifi::processors::GetFile::IgnoreHiddenFile, "false");
+
+  test_controller.runSession();
 
   REQUIRE(LogTestController::getInstance().contains("Logged 2 flow files"));
```
Comment (fgerlits) [inline extensions/standard-processors/tests/unit/GetFileTests.cpp:227]: I think this test, and some of the others, too, would be clearer if we also checked the file names of the files expected to be read, eg:
```suggestion
  REQUIRE(LogTestController::getInstance().contains("Logged 2 flow files"));
  REQUIRE(LogTestController::getInstance().contains("key:filename value:test.txt"));
  REQUIRE(LogTestController::getInstance().contains("key:filename value:.test.txt"));
```

**Review by fgerlits — COMMENTED** [review]
LGTM, with some minor comments

File: extensions/standard-processors/tests/unit/GetFileTests.cpp [hunk 2]
```diff
@@ -140,37 +142,103 @@ TEST_CASE("GetFile keeps the source file if KeepSourceFile is true") {
...
+
+    REQUIRE(LogTestController::getInstance().contains("Logged 3 flow files"));
+  }
+}
+
+TEST_CASE("Only older files are read when MinAge property is set", "[getFileProperty]") {
+  GetFileTestController test_controller;
+  test_controller.setProperty(minifi::processors::GetFile::MinAge, "1 hour");
+
+  utils::file::FileUtils::set_last_write_time(test_controller.getInputFilePath(), 936860949);
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply extensions/standard-processors/tests/unit/GetFileTests.cpp]: Updated in 447421035d07ab224a1d50855739a83d76417542

File: extensions/standard-processors/tests/unit/GetFileTests.cpp (hunk 1 above)
Comment (lordgamez) [reply extensions/standard-processors/tests/unit/GetFileTests.cpp:244]: Updated in 447421035d07ab224a1d50855739a83d76417542

**Review by fgerlits — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**